### DISCUSSION TITLE
How to make an object have multiple material properties？

### DISCUSSION THREAD

**Discussion by TeoREN0217** [SofaDefrost/SoftRobots#277] [Help - Programming with SoftRobots] [answered]
Hi everyone,

I'm currently working on constructing a robot with multiple layers, but I've encountered a limitation: when importing the robot mesh model, it appears I can only apply a single material definition. I'm exploring the possibility of creating an object with multiple materials and was considering capturing some mesh nodes to redefine them. However, I'm unsure if there's an efficient method to select these nodes. As far as I'm aware, SOFA presently offers a box selection tool for this purpose. Is there a better strategy to manage multi-material objects in such environments?

Also, could we use SOFACuda for Softrobots plugins now?

I'm looking forward to your reply!

Best,
Teo

**Comment by EulalieCoevoet** [comment]
Hello Teo, 

You can create meshes representing your layers (be careful that the volume mesh follows those boundaries) and try the component `MeshROI`: "Finds the primitives (vertex/edge/triangle/tetrahedron) inside a given mesh"

As for SOFACuda for the plugin SoftRobots, I've never tried it but there's a cmake option `PLUGIN_SOFTROBOTS_CUDA`. 

Eulalie

**Reply by TeoREN0217** [reply]
Hi Eulalie,

Thank so much for your reply! Sorry for my late reply. About the component 'MeshROI', if I want to add some layers, should I just mesh the main object and layers separately and using the 'MeshROI' to get the tetrahedron, then using 'TetrahedronFEMForceField' to define the meshROI parts' materials? I'm not sure if the pipeline is right. Or do we have some demo of the how using meshROI? 

Thank you for your reply again and looking forward to your reply!

Best,
Teo

**Reply by EulalieCoevoet** [reply, marked as answer]
Yes that's the idea. [Here's an example](https://github.com/sofa-framework/sofa/blob/master/examples/Component/Engine/Select/MeshROI.scn).

**Reply by TeoREN0217** [reply]
Thank you so much for your help!
